cmd/eksctl-anywhere/cmd/validate.go

```diff
@@ -0,0 +1,166 @@
+	validateCmd.Flags().BoolVar(&valOpt.skipIpCheck, "skip-ip-check", false, "Skip check for whether cluster control plane ip is in use")
+	validateCmd.Flags().StringVar(&valOpt.bundlesOverride, "bundles-override", "", "Override default Bundles manifest (not recommended)")
+	validateCmd.Flags().StringVar(&valOpt.managementKubeconfig, "kubeconfig", "", "Management cluster kubeconfig file")
+
+	if err := validateCmd.MarkFlagRequired("filename"); err != nil {
+		log.Fatalf("Error marking flag as required: %v", err)
+	}
+}
+
+func preRunValidate(cmd *cobra.Command, args []string) error {
+	cmd.Flags().VisitAll(func(flag *pflag.Flag) {
+		err := viper.BindPFlag(flag.Name, flag)
+		if err != nil {
+			log.Fatalf("Error initializing flags: %v", err)
+		}
+	})
+	return nil
+}
+
+func (valOpt *validateOptions) validateCluster(cmd *cobra.Command, _ []string) error {
+	ctx := cmd.Context()
+
+	validateCluster := workflows.NewValidate(ctx)
+
+	validateCluster.RunDockerValidations()
+
+	clusterConfig, err := cluster.ParseConfigFromFile(valOpt.fileName)
+	if err != nil {
+		return err
+	}
+
+	err = validateCluster.RunConfigValidations(clusterConfig)
+	if err != nil {
+		return err
+	}
+
+	if clusterConfig.Cluster.Spec.DatacenterRef.Kind == v1alpha1.TinkerbellDatacenterKind {
+		flag := cmd.Flags().Lookup(TinkerbellHardwareCSVFlagName)
+
+		// If no flag was returned there is a developer error as the flag has been removed
+		// from the program rendering it invalid.
+		if flag == nil {
+			panic("'hardwarefile' flag not configured")
+		}
+
+		if !viper.IsSet(TinkerbellHardwareCSVFlagName) || viper.GetString(TinkerbellHardwareCSVFlagName) == "" {
+			return fmt.Errorf("required flag \"%v\" not set", TinkerbellHardwareCSVFlagName)
+		}
+
+		if !validations.FileExists(cc.hardwareCSVPath) {
+			return fmt.Errorf("hardware config file %s does not exist", valOpt.hardwareCSVPath)
+		}
+	}
+
+	clusterSpec, err := newClusterSpec(valOpt.clusterOptions)
+	if err != nil {
+		return err
+	}
+
+	cliConfig := buildCliConfig(clusterSpec)
+	dirs, err := valOpt.directoriesToMount(clusterSpec, cliConfig)
+	if err != nil {
+		return err
+	}
+
+	tmpPath, err := os.MkdirTemp("./", "tmpValidate")
+	if err != nil {
+		return err
+	}
+
+	deps, err := dependencies.ForSpec(ctx, clusterSpec).WithExecutableMountDirs(dirs...).
+		WithWriterFolder(tmpPath).
+		WithKubectl().
+		WithProvider(valOpt.fileName, clusterSpec.Cluster, valOpt.skipIpCheck, valOpt.hardwareCSVPath, true, valOpt.tinkerbellBootstrapIP).
+		WithGitOpsFlux(clusterSpec.Cluster, clusterSpec.FluxConfig, cliConfig).
+		Build(ctx)
+	if err != nil {
+		return err
+	}
+	defer close(ctx, deps)
+
+	err = validateCluster.RunSpecValidations(clusterSpec, deps, cliConfig)
+	if err != nil {
+		return err
+	}
+
+	cleanup(deps, &err)
+	deps.Writer.CleanUp()
+	return err
+}
+
+func (valOpt *validateOptions) directoriesToMount(clusterSpec *cluster.Spec, cliConfig *config.CliConfig) ([]string, error) {
```

Comment:
is this copy paste from a different command?

Reply:
Yeah, this is taken directly from the createCluster cmd. Should I move it out into a separate function that both cmds can access?
https://github.com/aws/eks-anywhere/blob/9b1601f90b00ba5fbb39a0fd00cc06de53235c71/cmd/eksctl-anywhere/cmd/createcluster.go#L211